### ISSUE TITLE
Add (optional) allowed endpoints to global zones configuration

### ISSUE THREAD

**Issue by Mikesch-mp** [Icinga/icinga2#8182]
## Is your feature request related to a problem? Please describe.
If you create a global zone which should only exist on masters and satellites (on many different zones), which hold sensitive data like passwords etc. , and dont add this global zones on agents. The agent will log 

```
warning/ApiListener: Ignoring config update from endpoint 'some.host.fqdn' for unknown zone 'internal'.
```

With this information an user can easily add the zone on an agent and get the zone also synced to the agent and get the sensitive information out of this global zone.

## Describe the solution you'd like
There should be an option with an filter/array which endpoints are allowed to sync this global zone.
## Describe alternatives you've considered

There is no alternative, because if you have access to the logfiles and a litle bit knowledge on how icinga2 works, you can just add the global zone and see its content after a reload.

**Comment by Al2Klimov** [comment]
Shouldn't passwords be stored in services' custom variables?

**Comment by Mikesch-mp** [comment]
Yes they should, but not everything should be synced to agents. Global zones will be synced and if you know the name you can easily configure it on the agents and see its content.
For some services, api users etc. configuration which should not be synced and it is not needed to sync them to th agents it should be possible to disallow the sync even if (a evil admin/owner of an agent) reconfigure the agent to sync it.

Just think about enterprise installation with a lot of different owners/customers/compliance rules.

**Comment by Al2Klimov** [comment]
IMAO one should not put the respective services in global zones then.

**Comment by stevie-sy** [comment]
I would like to join in on this topic and I agree @Mikesch-mp with this feature-request. It's not only passwords! Think about the director with his "director-global" zone. Everything which is configured there with it you'll find in this zone (incl. passwords). Until there is no solution to limit this, such idea would be great. 
What I mean it's not necessary to sync the snmp password from a network device to a database server, because it's in the director-zone! 
I wrote some examples in this old bugs/feature-request:
- https://github.com/Icinga/icinga2/issues/6542
- https://github.com/Icinga/icingaweb2-module-director/issues/1589

**Comment by Mikesch-mp** [comment]
@Al2Klimov not everything is for non global zones. To make it easier for you to understand think of this.
For Agents that dont have access to the master nad only access to their satellites, you need an api user for the sattelites to enable create tickets for the agents, as for now i can put it in global zone, which is a bad idea or in the zone for the satellites (good idea because it doesnt get synced to agents). But i have to create the same api user for all zone, which is anoying if you have a few hundred zones. Better put the api users into a global zone which will be only synced to satellites, so you have only to define your api user once. Problem here is that the agent see this "unknown" zone in their log file and can get it synced by adding it to the agents zone.conf -> User/Customer/Spy can see passwords or other things they should not be able to see.

Hope this makes it a bit clearer for you.

Regards,
Carsten

**Comment by RincewindsHat** [comment]
Ping!
